nf_core/modules/module_test.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+"""
+The ModulesTest class runs the tests locally
+"""
+
+import logging
+import questionary
+import os
+import pytest
+import sys
+import rich
+
+import nf_core.utils
+from .modules_repo import ModulesRepo
+
+log = logging.getLogger(__name__)
+
+
+class ModulesTest(object):
+    """
+    Class to run module pytests.
+
+    ...
+
+    Attributes
+    ----------
+    module_name : str
+        name of the tool to run tests for
+    no_prompts : bool
+        flat indicating if prompts are used
+    pytest_args : tuple
+        additional arguments passed to pytest command
+
+    Methods
+    -------
+    run():
+        Run test steps
+    __check_inputs():
+        Check inputs. Ask for module_name if not provided and check that the directory exists
+    __set_profile():
+        Set software profile
+    __run_pytests(self):
+        Run pytest
+    """
+
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+        pytest_args="",
+    ):
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.pytest_args = pytest_args
+        self.module_dir = None
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self.__check_inputs()
+        self.__set_profile()
+        self.__run_pytests()
+
+    def __check_inputs(self):
+        """Do more complex checks about supplied flags."""
+
+        # Get the tool name if not specified
+        if self.module_name is None:
+            if self.no_prompts:
+                raise UserWarning(
+                    f"Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
+                )
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        self.module_dir = os.path.join("modules", *self.module_name.split("/"))
+
+        # First, sanity check that the module directory exists
+        if not os.path.isdir(self.module_dir):
+            raise UserWarning(f"Cannot find directory '{self.module_dir}'. Should be TOOL/SUBTOOL or TOOL")
+
+    def __set_profile(self):
+        """Set $PROFILE env variable.
+        The config expects $PROFILE and Nextflow fails if it's not set.
+        """
+        if os.environ.get("PROFILE") is None:
+            os.environ["PROFILE"] = ""
+            if self.no_prompts:
+                log.info(
+                    "Setting env var '$PROFILE' to an empty string as not set.\n"
```

Comment:
I would not abbreviate "the environment variable 'PROFILE'" to "env var '$PROFILE'". It might not be obvious for someone new to shells.